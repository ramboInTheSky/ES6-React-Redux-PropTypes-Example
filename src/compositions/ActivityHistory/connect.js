import { connect } from 'react-redux';
import path from 'ramda/src/path';

import { getActivityHistory, invalidateActivityHistory } from '../../ducks/activities';

export const mapStateToProps = state => {
  const {
    activityHistory: { heading, labels },
  } = state.dictionary;

  return {
    activityHistory: path(['activities', 'history'], state),
    heading,
    labels,
  };
};

export const mergeProps = (
  { arrears, tenants, vulnerabilityFlag, ...stateProps },
  { dispatch },
  ownProps
) => {
  const arrearsId = path(['match', 'params', 'arrearsId'], ownProps);

  return {
    ...stateProps,
    arrearsId,
    invalidateActivityHistory: () => dispatch(invalidateActivityHistory()),
    getActivityHistory: () => dispatch(getActivityHistory(arrearsId)),
    ...ownProps,
  };
};

export default connect(
  mapStateToProps,
  null,
  mergeProps
);
