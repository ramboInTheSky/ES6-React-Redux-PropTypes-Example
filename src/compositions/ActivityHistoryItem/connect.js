import { connect } from 'react-redux';
import path from 'ramda/src/path';

import activityNavigation from '../../util/activityNavigation';
import getHelpers from '../../util/stateHelpers';

import { updateRibbon } from '../../ducks/ribbon';
import { getActivity, getActivityHistory, invalidateActivityDetail } from '../../ducks/activities';

export const mapStateToProps = (state, ownProps) => {
  const { get, getBreadcrumbs } = getHelpers(state);

  const {
    activityHistory: { itemHeading: heading, labels, values, attachments: attachmentsHeading },
    arrearsDetails: { heading: arrearsDetailsHeading },
    genericErrorText,
    paymentPlan: { installmentArrangementFormat },
  } = state.dictionary;

  const referralLabels = path(['dictionary', 'referral'], state);
  const pauseLabels = path(['dictionary', 'pause'], state);
  const paymentLabels = path(['dictionary', 'payment', 'labels'], state);
  const paymentPlanLabels = path(['dictionary', 'paymentPlan', 'labels'], state);
  const legalReferralLabels = path(['dictionary', 'legalReferral'], state);

  const arrearsId = path(['match', 'params', 'arrearsId'], ownProps);
  const itemId = path(['match', 'params', 'itemId'], ownProps);
  const redirectToDetailsPage = () => ownProps.history.push(`/arrears-details/${arrearsId}`);
  const goToHistoryItem = historyId =>
    ownProps.history.push(`/arrears-details/${arrearsId}/history/${historyId}`);
  const activities = get(['activities', 'history'], []);
  const attachments = get(['activities', 'attachments']);
  const attachmentsLoading = get(['activities', 'attachmentsLoading'], []);
  const attachmentsError = get(['activities', 'attachmentsError'], []);
  const activityHistoryNavigation = activityNavigation(activities, itemId);

  return {
    activities,
    activityHistoryNavigation,
    genericErrorText,
    attachmentsHeading,
    activityDetail: get(['activities', 'detail']),
    activityError: get(['activities', 'error']),
    activityLoading: get(['activities', 'loading']),
    arrearsId,
    breadcrumb: getBreadcrumbs(
      heading,
      {},
      {
        label: arrearsDetailsHeading,
        to: `/arrears-details/${arrearsId}`,
      }
    ),
    canGoNext: !activityHistoryNavigation.isLast(),
    canGoPrevious: !activityHistoryNavigation.isFirst(),
    heading,
    goToHistoryItem,
    itemId,
    labels,
    redirectToDetailsPage,
    values,
    referralLabels,
    pauseLabels,
    paymentLabels,
    paymentPlanLabels: { installmentArrangementFormat, ...paymentPlanLabels },
    legalReferralLabels,
    attachments,
    attachmentsError,
    attachmentsLoading,
  };
};

export const mergeProps = (
  {
    activities,
    activityDetail,
    activityHistoryNavigation,
    arrearsId,
    breadcrumb,
    heading,
    goToHistoryItem,
    itemId,
    redirectToDetailsPage,
    ...stateProps
  },
  { dispatch },
  ownProps
) => ({
  ...stateProps,
  activities,
  itemId,
  activityDetail,
  getActivityById: id => {
    const activityHistoryItem = activities.find(activity => activity.id === id);
    dispatch(getActivity(activityHistoryItem)).catch(() => {});
  },
  getActivityHistory: () => {
    dispatch(getActivityHistory(arrearsId));
  },
  invalidateActivityDetail: () => dispatch(invalidateActivityDetail()),
  onBack: () => redirectToDetailsPage(),
  onNext: () => {
    const nextHistoryId = activityHistoryNavigation.getNextHistoryId();
    if (nextHistoryId) {
      goToHistoryItem(nextHistoryId);
    }
  },
  onPrevious: () => {
    const prevHistoryId = activityHistoryNavigation.getPreviousHistoryId();
    if (prevHistoryId) {
      goToHistoryItem(prevHistoryId);
    }
  },
  updatePageHeader: () => dispatch(updateRibbon({ title: heading, breadcrumb })),
  ...ownProps,
});

export default connect(
  mapStateToProps,
  null,
  mergeProps
);
