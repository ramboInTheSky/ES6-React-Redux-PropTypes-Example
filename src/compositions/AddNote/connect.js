import { connect } from 'react-redux';
import path from 'ramda/src/path';

import getHelpers from '../../util/stateHelpers';
import { addNotification } from '../../ducks/notifications';
import { updateRibbon } from '../../ducks/ribbon';
import { addNoteToCase, clearNoteError } from '../../ducks/notes';

export const mapStateToProps = (state, ownProps) => {
  const { get, getBreadcrumbs } = getHelpers(state);
  const {
    addNote: { addNote, formError, notification, ...text },
    arrearsDetails: { heading: arrearsDetailsHeading },
  } = state.dictionary;

  const arrearsId = path(['match', 'params', 'arrearsId'], ownProps);
  const returnUrl = `/arrears-details/${arrearsId}`;
  const redirectToDetailsPage = () => ownProps.history.push(returnUrl);

  const redirectToAttachmentsPage = id =>
    ownProps.history.push(`${returnUrl}/note/${id}/attachments?redirectTo=${returnUrl}`);

  return {
    addNote,
    arrearsId,
    breadcrumb: getBreadcrumbs(
      addNote,
      {},
      {
        label: arrearsDetailsHeading,
        to: `/arrears-details/${arrearsId}`,
      }
    ),
    error: get(['notes', 'error']),
    isLoading: get(['notes', 'loading']),
    formError,
    heading: addNote,
    notification,
    redirectToDetailsPage,
    redirectToAttachmentsPage,
    ...text,
  };
};

export const mergeProps = (
  {
    arrearsId,
    breadcrumb,
    heading,
    notification,
    redirectToAttachmentsPage,
    redirectToDetailsPage,
    ...stateProps
  },
  { dispatch },
  ownProps
) => ({
  ...stateProps,
  arrearsId,
  clearError: () => dispatch(clearNoteError()),
  onBack: redirectToDetailsPage,
  onSubmit: (note, wantsToAttachFiles) =>
    dispatch(addNoteToCase(arrearsId, note))
      .then(id => {
        dispatch(
          addNotification({
            dataBddPrefix: 'addNote',
            lines: [{ text: notification.line1 }],
            icon: 'success',
            notificationType: 'confirmation',
            title: notification.title,
          })
        );

        if (wantsToAttachFiles) {
          redirectToAttachmentsPage(id);
        } else {
          redirectToDetailsPage();
        }
      })
      .catch(() => {}),
  updatePageHeader: () => dispatch(updateRibbon({ title: heading, breadcrumb })),
  ...ownProps,
});

export default connect(
  mapStateToProps,
  null,
  mergeProps
);
