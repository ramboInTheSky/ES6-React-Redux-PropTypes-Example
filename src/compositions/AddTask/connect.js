import { connect } from 'react-redux';
import path from 'ramda/src/path';
import { formatting } from 'nhh-styles';
import getHelpers from '../../util/stateHelpers';
import { addNotification } from '../../ducks/notifications';
import { updateRibbon } from '../../ducks/ribbon';
import { clearTaskError, taskError, addTask as AddTaskItem } from '../../ducks/tasks';
import { clearSearch, search } from '../../ducks/housingofficersearch';

export const mapStateToProps = (state, ownProps) => {
  const { get, getBreadcrumbs } = getHelpers(state);
  const {
    addTask,
    addTask: { addTask: heading, notification, formError, serverError },
    arrearsDetails: { heading: arrearsDetailsHeading },
  } = state.dictionary;

  const arrearsId = path(['match', 'params', 'arrearsId'], ownProps);
  const redirectToDetailsPage = () => ownProps.history.push(`/arrears-details/${arrearsId}`);

  return {
    addTask,
    arrearsId,
    breadcrumb: getBreadcrumbs(
      heading,
      {},
      {
        label: arrearsDetailsHeading,
        to: `/arrears-details/${arrearsId}`,
      }
    ),
    loading: get(['tasks', 'loading']),
    housingOfficerSearchResults: get(['housingofficersearch', 'searchResults']) || [],
    error: get(['tasks', 'error']),
    userId: get(['user', 'profile', 'id']),
    loggedInUserId: get(['user', 'profile', 'id']),
    userFullName: get(['user', 'profile', 'fullname']),
    notification,
    heading,
    redirectToDetailsPage,
    errorMessage: get(['tasks', 'isFormError']) ? formError : serverError,
  };
};

export const mergeProps = (
  {
    arrearsId,
    breadcrumb,
    heading,
    housingOfficerSearchResults,
    notification,
    redirectToDetailsPage,
    userId,
    ...stateProps
  },
  { dispatch },
  ownProps
) => ({
  ...stateProps,
  arrearsId,
  housingOfficerSearchResults,
  clearUserSearch: () => dispatch(clearSearch()),
  onBack: () => redirectToDetailsPage(),
  onDisplayFormError: () => dispatch(taskError(true)),
  onSubmit: ({ title, description, date, ownerId }) => {
    dispatch(AddTaskItem(arrearsId, title, description, date, ownerId, userId))
      .then(() => {
        dispatch(
          addNotification({
            dataBddPrefix: 'addTask',
            lines: [{ text: `${title} ${formatting.formatDate(date)}` }],
            icon: 'success',
            notificationType: 'confirmation',
            title: notification.title,
          })
        );
        redirectToDetailsPage();
      })
      .catch(() => {});
  },
  onSearch: term => {
    if (term && term.length >= 3) {
      dispatch(search(term));
    } else if (term.length === 0 && !!housingOfficerSearchResults.length) {
      dispatch(clearSearch());
    }
  },
  onUnmount: () => {
    dispatch(clearTaskError());
    dispatch(clearSearch());
  },
  updatePageHeader: () => dispatch(updateRibbon({ title: heading, breadcrumb })),
  // TODO: remove this next line. By the looks of it is not needed anymore. However, I'm not 100% sure.
  // So I'm just leaving it here in case this later becomes a bug, in QA.
  ...ownProps,
});

export default connect(
  mapStateToProps,
  null,
  mergeProps
);
