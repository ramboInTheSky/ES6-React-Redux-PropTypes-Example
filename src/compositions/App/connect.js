import compose from 'ramda/src/compose';
import pickAll from 'ramda/src/pickAll';
import { connect } from 'react-redux';

import { displayLogin, logout } from '../../ducks/user';
import { setEntryPoint } from '../../ducks/navigation';
import { hideModal } from '../../ducks/modal';

export const mapStateToProps = pickAll(['user', 'ribbon', 'modal']);

export const mergeProps = (stateProps, { dispatch }, ownProps) => ({
  ...stateProps,
  login: () => dispatch(displayLogin()),
  logout: () => dispatch(logout()),
  modal: {
    ...stateProps.modal,
    onClose: compose(
      dispatch,
      hideModal
    ),
  },
  setEntryPoint: path => dispatch(setEntryPoint(path)),
  ...ownProps,
});

export default connect(
  mapStateToProps,
  null,
  mergeProps
);
