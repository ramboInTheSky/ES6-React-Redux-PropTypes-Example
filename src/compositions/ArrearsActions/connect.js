import { connect } from 'react-redux';
import path from 'ramda/src/path';
import format from 'string-format';

import getInternalRoutes from '../../constants/internalRoutes';
import getHelpers from '../../util/stateHelpers';
import { payments } from '../../constants/routes';
import { ACTIVE, DRAFT } from '../../constants/pause';
import { SYSTEM_PAUSED } from '../../constants/arrears';

export const mapStateToProps = (state, ownProps) => {
  const { get } = getHelpers(state);
  const arrearsId = path(['match', 'params', 'arrearsId'], ownProps);
  const arrears = get(['arrears', 'items']);
  const currentArrear = arrears.find(arrear => arrear.id === arrearsId) || {};
  const isArrearSystemPaused = currentArrear.status && currentArrear.status === SYSTEM_PAUSED;
  const isManager = get(['user', 'userType']) === 'manager';

  const internalRouter = getInternalRoutes(currentArrear, isManager);

  const {
    addNote: { addNote },
    addTask: { addTask },
    correspondence: { sendCorrespondence },
    interaction: { addInteraction },
    legalCaseReferral: { create, review },
    pauseArrearsCase: { pauseArrearsCase, reviewPauseRequest, updatePauseRequest },
    paymentPlan: { createPaymentPlan, viewPaymentPlan },
    referArrearsCase: { heading2: tenancySustainmentReferral },
    serveNosp: { serveNospEarly },
    takePayment: { takePayment },
  } = state.dictionary;

  // NOSP button
  const serveNosp = currentArrear.canServeNosp
    ? [
        {
          dataBdd: 'serveNosp',
          text: serveNospEarly,
          link: internalRouter.arrearsDetailNOSPModal,
        },
      ]
    : [];

  // Pause button
  const pauses = path(['pauseSummary'], currentArrear) || [];
  const pauseAwaitingApproval = pauses.filter(pause => pause.status === DRAFT)[0];
  const activePause = pauses.filter(pause => pause.status === ACTIVE)[0] || pauseAwaitingApproval;
  let pauseButtontext = activePause ? updatePauseRequest : pauseArrearsCase;
  if (pauseAwaitingApproval && isManager) {
    pauseButtontext = reviewPauseRequest;
  }

  const pauseRequest = isArrearSystemPaused
    ? []
    : [
        {
          dataBdd: 'pauseArrearsCase',
          text: pauseButtontext,
          link: internalRouter.pause,
        },
      ];

  // Legal referral button
  const canCreateLegalReferral = path(
    ['legalReferralCase', 'canCreateLegalReferral'],
    currentArrear
  );
  const longName = path(['legalReferralCase', 'formName'], currentArrear);
  const legalReferralActionName = longName && longName.split('|')[0].trim();
  const currentLegalReferralAction = path(['legalReferral', 'caseId'], currentArrear);

  const legalReferralAction =
    legalReferralActionName && canCreateLegalReferral
      ? [
          {
            dataBdd: 'legalCaseReferral',
            text: format(currentLegalReferralAction ? review : create, {
              type: legalReferralActionName,
            }),
            link: internalRouter.legalReferral,
          },
        ]
      : [];

  return {
    actions: [
      {
        dataBdd: 'addInteraction',
        text: addInteraction,
        link: internalRouter.interaction,
      },
      {
        dataBdd: 'sendCorrespondence',
        text: sendCorrespondence,
        link: internalRouter.sendCorrespondence,
      },
      ...serveNosp,
      {
        dataBdd: 'takePayment',
        text: takePayment,
        href: `${payments}/setup-payment?tenancyId=${currentArrear.tenancyId}&arrearsId=${
          currentArrear.id
        }`,
      },
      {
        dataBdd: 'paymentPlan',
        text: !currentArrear.paymentPlan ? createPaymentPlan : viewPaymentPlan,
        link: internalRouter.paymentPlan,
      },
      {
        dataBdd: 'tenancySustainmentReferral',
        text: tenancySustainmentReferral,
        link: internalRouter.tenancySustainmentReferral,
      },
      {
        dataBdd: 'addTask',
        text: addTask,
        link: internalRouter.addTask,
      },
      {
        dataBdd: 'addNote',
        text: addNote,
        link: internalRouter.addNote,
      },
      ...pauseRequest,
      ...legalReferralAction,
    ],
  };
};

export default connect(mapStateToProps);
