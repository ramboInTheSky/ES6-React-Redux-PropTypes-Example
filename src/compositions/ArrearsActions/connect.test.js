import { mapStateToProps } from './connect';

import legalReferralStatus from '../../constants/legalReferral';
import { SYSTEM_PAUSED } from '../../constants/arrears';
import state from '../../../__mocks__/state';

describe('ArrearsDetail connector', () => {
  let mockState;
  let ownProps;
  let mapStateToPropsResult;

  beforeEach(() => {
    mockState = state();

    ownProps = {
      match: {
        params: {
          arrearsId: 'foo',
        },
      },
    };
  });

  describe('mapStateToProps', () => {
    it('generates the correct props', () => {
      mapStateToPropsResult = mapStateToProps(mockState, ownProps);
      expect(mapStateToPropsResult).toMatchSnapshot();
    });

    describe('Hide pause actions', () => {
      it('generates the correct props when status is SYSTEM_PAUSED', () => {
        const newState = {
          ...mockState,
          arrears: {
            ...mockState.arrears,
            items: [
              {
                ...mockState.arrears.items[0],
                status: SYSTEM_PAUSED,
              },
            ],
          },
        };
        mapStateToPropsResult = mapStateToProps(newState, ownProps);
        expect(mapStateToPropsResult).toMatchSnapshot();
      });
    });

    describe('serve NOSP', () => {
      it('generates the correct props when canServeNosp is enabled', () => {
        const newState = {
          ...mockState,
          arrears: {
            ...mockState.arrears,
            items: [
              {
                ...mockState.arrears.items[0],
                canServeNosp: true,
              },
            ],
          },
        };
        mapStateToPropsResult = mapStateToProps(newState, ownProps);
        expect(mapStateToPropsResult).toMatchSnapshot();
      });
    });

    describe('Payment Plan', () => {
      it('generates the correct props when a payment plan exists', () => {
        const newState = {
          ...mockState,
          arrears: {
            ...mockState.arrears,
            items: [
              {
                ...mockState.arrears.items[0],
                paymentPlan: { id: 'abc123' },
              },
            ],
          },
        };
        mapStateToPropsResult = mapStateToProps(newState, ownProps);
        expect(mapStateToPropsResult).toMatchSnapshot();
      });
    });

    describe('create pause', () => {
      it('generates the correct props when there is an active pause', () => {
        const newState = {
          ...mockState,
          arrears: {
            ...mockState.arrears,
            items: [
              {
                ...mockState.arrears.items[0],
                pauseSummary: [{ id: 'abc123', status: 'Draft' }],
              },
            ],
          },
        };
        mapStateToPropsResult = mapStateToProps(newState, ownProps);
        expect(mapStateToPropsResult).toMatchSnapshot();
      });

      it('generates the correct props when there is a pause awaiting approval and user is an HO', () => {
        const newState = {
          ...mockState,
          arrears: {
            ...mockState.arrears,
            items: [
              {
                ...mockState.arrears.items[0],
                pauseSummary: [{ id: 'def456', status: 'Active' }],
              },
            ],
          },
        };
        mapStateToPropsResult = mapStateToProps(newState, ownProps);
        expect(mapStateToPropsResult).toMatchSnapshot();
      });

      it('generates the correct props when there is a pause awaiting approval and user is an HM', () => {
        const newState = {
          ...mockState,
          arrears: {
            ...mockState.arrears,
            items: [
              {
                ...mockState.arrears.items[0],
                pauseSummary: [{ id: 'ghi789', status: 'Active' }],
              },
            ],
          },
          user: { userType: 'manager' },
        };
        mapStateToPropsResult = mapStateToProps(newState, ownProps);
        expect(mapStateToPropsResult).toMatchSnapshot();
      });
    });

    describe('Legal referral action', () => {
      it('generates the correct props when there is a legal referring waiting for submission', () => {
        const newState = {
          ...mockState,
          arrears: {
            ...mockState.arrears,
            items: [
              {
                ...mockState.arrears.items[0],
                legalReferralCase: {
                  ...mockState.arrears.items[0].legalReferralCase,
                  canCreateLegalReferral: true,
                  lastSubmissionId: 'abc123',
                },
              },
            ],
          },
        };
        mapStateToPropsResult = mapStateToProps(newState, ownProps);
        expect(mapStateToPropsResult).toMatchSnapshot();
      });

      it('generates the correct props when there is a legal referring waiting for manager approval', () => {
        const newState = {
          ...mockState,
          arrears: {
            ...mockState.arrears,
            items: [
              {
                ...mockState.arrears.items[0],
                legalReferral: {
                  caseId: 'case id',
                },
                legalReferralCase: {
                  ...mockState.arrears.items[0].legalReferralCase,
                  canCreateLegalReferral: true,
                  lastSubmissionId: 'abc123',
                  status: legalReferralStatus.requiresApproval,
                },
              },
            ],
          },
        };
        mapStateToPropsResult = mapStateToProps(newState, ownProps);
        expect(mapStateToPropsResult).toMatchSnapshot();
      });
    });
  });
});
