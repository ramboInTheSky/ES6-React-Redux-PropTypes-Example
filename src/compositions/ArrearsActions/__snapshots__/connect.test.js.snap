// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrearsDetail connector mapStateToProps Hide pause actions generates the correct props when status is SYSTEM_PAUSED 1`] = `
Object {
  "actions": Array [
    Object {
      "dataBdd": "addInteraction",
      "link": "/arrears-details/foo/interaction/create",
      "text": "Add interaction",
    },
    Object {
      "dataBdd": "sendCorrespondence",
      "link": "/arrears-details/foo/send-correspondence/create",
      "text": "Send correspondence",
    },
    Object {
      "dataBdd": "takePayment",
      "href": "#{Payments.Core.BaseURL}/setup-payment?tenancyId=ABC123&arrearsId=foo",
      "text": "New payment",
    },
    Object {
      "dataBdd": "paymentPlan",
      "link": "/arrears-details/foo/payment-plan/create",
      "text": "New payment plan",
    },
    Object {
      "dataBdd": "tenancySustainmentReferral",
      "link": "/arrears-details/foo/refer-arrears-case/create",
      "text": "Tenancy sustainment referral",
    },
    Object {
      "dataBdd": "addTask",
      "link": "/arrears-details/foo/task/create",
      "text": "Add Task",
    },
    Object {
      "dataBdd": "addNote",
      "link": "/arrears-details/foo/note/create",
      "text": "Add note",
    },
  ],
}
`;

exports[`ArrearsDetail connector mapStateToProps Legal referral action generates the correct props when there is a legal referring waiting for manager approval 1`] = `
Object {
  "actions": Array [
    Object {
      "dataBdd": "addInteraction",
      "link": "/arrears-details/foo/interaction/create",
      "text": "Add interaction",
    },
    Object {
      "dataBdd": "sendCorrespondence",
      "link": "/arrears-details/foo/send-correspondence/create",
      "text": "Send correspondence",
    },
    Object {
      "dataBdd": "takePayment",
      "href": "#{Payments.Core.BaseURL}/setup-payment?tenancyId=ABC123&arrearsId=foo",
      "text": "New payment",
    },
    Object {
      "dataBdd": "paymentPlan",
      "link": "/arrears-details/foo/payment-plan/create",
      "text": "New payment plan",
    },
    Object {
      "dataBdd": "tenancySustainmentReferral",
      "link": "/arrears-details/foo/refer-arrears-case/create",
      "text": "Tenancy sustainment referral",
    },
    Object {
      "dataBdd": "addTask",
      "link": "/arrears-details/foo/task/create",
      "text": "Add Task",
    },
    Object {
      "dataBdd": "addNote",
      "link": "/arrears-details/foo/note/create",
      "text": "Add note",
    },
    Object {
      "dataBdd": "pauseArrearsCase",
      "link": "/arrears-details/foo/pause-arrears-case/create",
      "text": "Pause arrears case",
    },
    Object {
      "dataBdd": "legalCaseReferral",
      "link": "/arrears-details/foo/legal-case-referral/case id/approve",
      "text": "Review Friendly Form Name",
    },
  ],
}
`;

exports[`ArrearsDetail connector mapStateToProps Legal referral action generates the correct props when there is a legal referring waiting for submission 1`] = `
Object {
  "actions": Array [
    Object {
      "dataBdd": "addInteraction",
      "link": "/arrears-details/foo/interaction/create",
      "text": "Add interaction",
    },
    Object {
      "dataBdd": "sendCorrespondence",
      "link": "/arrears-details/foo/send-correspondence/create",
      "text": "Send correspondence",
    },
    Object {
      "dataBdd": "takePayment",
      "href": "#{Payments.Core.BaseURL}/setup-payment?tenancyId=ABC123&arrearsId=foo",
      "text": "New payment",
    },
    Object {
      "dataBdd": "paymentPlan",
      "link": "/arrears-details/foo/payment-plan/create",
      "text": "New payment plan",
    },
    Object {
      "dataBdd": "tenancySustainmentReferral",
      "link": "/arrears-details/foo/refer-arrears-case/create",
      "text": "Tenancy sustainment referral",
    },
    Object {
      "dataBdd": "addTask",
      "link": "/arrears-details/foo/task/create",
      "text": "Add Task",
    },
    Object {
      "dataBdd": "addNote",
      "link": "/arrears-details/foo/note/create",
      "text": "Add note",
    },
    Object {
      "dataBdd": "pauseArrearsCase",
      "link": "/arrears-details/foo/pause-arrears-case/create",
      "text": "Pause arrears case",
    },
    Object {
      "dataBdd": "legalCaseReferral",
      "link": "/arrears-details/foo/legal-case-referral/create",
      "text": "Friendly Form Name",
    },
  ],
}
`;

exports[`ArrearsDetail connector mapStateToProps Payment Plan generates the correct props when a payment plan exists 1`] = `
Object {
  "actions": Array [
    Object {
      "dataBdd": "addInteraction",
      "link": "/arrears-details/foo/interaction/create",
      "text": "Add interaction",
    },
    Object {
      "dataBdd": "sendCorrespondence",
      "link": "/arrears-details/foo/send-correspondence/create",
      "text": "Send correspondence",
    },
    Object {
      "dataBdd": "takePayment",
      "href": "#{Payments.Core.BaseURL}/setup-payment?tenancyId=ABC123&arrearsId=foo",
      "text": "New payment",
    },
    Object {
      "dataBdd": "paymentPlan",
      "link": "/arrears-details/foo/payment-plan/abc123",
      "text": "View payment plan",
    },
    Object {
      "dataBdd": "tenancySustainmentReferral",
      "link": "/arrears-details/foo/refer-arrears-case/create",
      "text": "Tenancy sustainment referral",
    },
    Object {
      "dataBdd": "addTask",
      "link": "/arrears-details/foo/task/create",
      "text": "Add Task",
    },
    Object {
      "dataBdd": "addNote",
      "link": "/arrears-details/foo/note/create",
      "text": "Add note",
    },
    Object {
      "dataBdd": "pauseArrearsCase",
      "link": "/arrears-details/foo/pause-arrears-case/create",
      "text": "Pause arrears case",
    },
  ],
}
`;

exports[`ArrearsDetail connector mapStateToProps create pause generates the correct props when there is a pause awaiting approval and user is an HM 1`] = `
Object {
  "actions": Array [
    Object {
      "dataBdd": "addInteraction",
      "link": "/arrears-details/foo/interaction/create",
      "text": "Add interaction",
    },
    Object {
      "dataBdd": "sendCorrespondence",
      "link": "/arrears-details/foo/send-correspondence/create",
      "text": "Send correspondence",
    },
    Object {
      "dataBdd": "takePayment",
      "href": "#{Payments.Core.BaseURL}/setup-payment?tenancyId=ABC123&arrearsId=foo",
      "text": "New payment",
    },
    Object {
      "dataBdd": "paymentPlan",
      "link": "/arrears-details/foo/payment-plan/create",
      "text": "New payment plan",
    },
    Object {
      "dataBdd": "tenancySustainmentReferral",
      "link": "/arrears-details/foo/refer-arrears-case/create",
      "text": "Tenancy sustainment referral",
    },
    Object {
      "dataBdd": "addTask",
      "link": "/arrears-details/foo/task/create",
      "text": "Add Task",
    },
    Object {
      "dataBdd": "addNote",
      "link": "/arrears-details/foo/note/create",
      "text": "Add note",
    },
    Object {
      "dataBdd": "pauseArrearsCase",
      "link": "/arrears-details/foo/pause-arrears-case/ghi789/update",
      "text": "Update/cancel pause",
    },
  ],
}
`;

exports[`ArrearsDetail connector mapStateToProps create pause generates the correct props when there is a pause awaiting approval and user is an HO 1`] = `
Object {
  "actions": Array [
    Object {
      "dataBdd": "addInteraction",
      "link": "/arrears-details/foo/interaction/create",
      "text": "Add interaction",
    },
    Object {
      "dataBdd": "sendCorrespondence",
      "link": "/arrears-details/foo/send-correspondence/create",
      "text": "Send correspondence",
    },
    Object {
      "dataBdd": "takePayment",
      "href": "#{Payments.Core.BaseURL}/setup-payment?tenancyId=ABC123&arrearsId=foo",
      "text": "New payment",
    },
    Object {
      "dataBdd": "paymentPlan",
      "link": "/arrears-details/foo/payment-plan/create",
      "text": "New payment plan",
    },
    Object {
      "dataBdd": "tenancySustainmentReferral",
      "link": "/arrears-details/foo/refer-arrears-case/create",
      "text": "Tenancy sustainment referral",
    },
    Object {
      "dataBdd": "addTask",
      "link": "/arrears-details/foo/task/create",
      "text": "Add Task",
    },
    Object {
      "dataBdd": "addNote",
      "link": "/arrears-details/foo/note/create",
      "text": "Add note",
    },
    Object {
      "dataBdd": "pauseArrearsCase",
      "link": "/arrears-details/foo/pause-arrears-case/def456/update",
      "text": "Update/cancel pause",
    },
  ],
}
`;

exports[`ArrearsDetail connector mapStateToProps create pause generates the correct props when there is an active pause 1`] = `
Object {
  "actions": Array [
    Object {
      "dataBdd": "addInteraction",
      "link": "/arrears-details/foo/interaction/create",
      "text": "Add interaction",
    },
    Object {
      "dataBdd": "sendCorrespondence",
      "link": "/arrears-details/foo/send-correspondence/create",
      "text": "Send correspondence",
    },
    Object {
      "dataBdd": "takePayment",
      "href": "#{Payments.Core.BaseURL}/setup-payment?tenancyId=ABC123&arrearsId=foo",
      "text": "New payment",
    },
    Object {
      "dataBdd": "paymentPlan",
      "link": "/arrears-details/foo/payment-plan/create",
      "text": "New payment plan",
    },
    Object {
      "dataBdd": "tenancySustainmentReferral",
      "link": "/arrears-details/foo/refer-arrears-case/create",
      "text": "Tenancy sustainment referral",
    },
    Object {
      "dataBdd": "addTask",
      "link": "/arrears-details/foo/task/create",
      "text": "Add Task",
    },
    Object {
      "dataBdd": "addNote",
      "link": "/arrears-details/foo/note/create",
      "text": "Add note",
    },
    Object {
      "dataBdd": "pauseArrearsCase",
      "link": "/arrears-details/foo/pause-arrears-case/abc123/update",
      "text": "Update/cancel pause",
    },
  ],
}
`;

exports[`ArrearsDetail connector mapStateToProps generates the correct props 1`] = `
Object {
  "actions": Array [
    Object {
      "dataBdd": "addInteraction",
      "link": "/arrears-details/foo/interaction/create",
      "text": "Add interaction",
    },
    Object {
      "dataBdd": "sendCorrespondence",
      "link": "/arrears-details/foo/send-correspondence/create",
      "text": "Send correspondence",
    },
    Object {
      "dataBdd": "takePayment",
      "href": "#{Payments.Core.BaseURL}/setup-payment?tenancyId=ABC123&arrearsId=foo",
      "text": "New payment",
    },
    Object {
      "dataBdd": "paymentPlan",
      "link": "/arrears-details/foo/payment-plan/create",
      "text": "New payment plan",
    },
    Object {
      "dataBdd": "tenancySustainmentReferral",
      "link": "/arrears-details/foo/refer-arrears-case/create",
      "text": "Tenancy sustainment referral",
    },
    Object {
      "dataBdd": "addTask",
      "link": "/arrears-details/foo/task/create",
      "text": "Add Task",
    },
    Object {
      "dataBdd": "addNote",
      "link": "/arrears-details/foo/note/create",
      "text": "Add note",
    },
    Object {
      "dataBdd": "pauseArrearsCase",
      "link": "/arrears-details/foo/pause-arrears-case/create",
      "text": "Pause arrears case",
    },
  ],
}
`;

exports[`ArrearsDetail connector mapStateToProps serve NOSP generates the correct props when canServeNosp is enabled 1`] = `
Object {
  "actions": Array [
    Object {
      "dataBdd": "addInteraction",
      "link": "/arrears-details/foo/interaction/create",
      "text": "Add interaction",
    },
    Object {
      "dataBdd": "sendCorrespondence",
      "link": "/arrears-details/foo/send-correspondence/create",
      "text": "Send correspondence",
    },
    Object {
      "dataBdd": "serveNosp",
      "link": "/arrears-details/foo?modal=serveNOSP",
      "text": "Serve NOSP early",
    },
    Object {
      "dataBdd": "takePayment",
      "href": "#{Payments.Core.BaseURL}/setup-payment?tenancyId=ABC123&arrearsId=foo",
      "text": "New payment",
    },
    Object {
      "dataBdd": "paymentPlan",
      "link": "/arrears-details/foo/payment-plan/create",
      "text": "New payment plan",
    },
    Object {
      "dataBdd": "tenancySustainmentReferral",
      "link": "/arrears-details/foo/refer-arrears-case/create",
      "text": "Tenancy sustainment referral",
    },
    Object {
      "dataBdd": "addTask",
      "link": "/arrears-details/foo/task/create",
      "text": "Add Task",
    },
    Object {
      "dataBdd": "addNote",
      "link": "/arrears-details/foo/note/create",
      "text": "Add note",
    },
    Object {
      "dataBdd": "pauseArrearsCase",
      "link": "/arrears-details/foo/pause-arrears-case/create",
      "text": "Pause arrears case",
    },
  ],
}
`;
